lib/bolt/module_installer/puppetfile/forge_module.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'semantic_puppet'
+require 'set'
+
+require_relative '../../error'
+
+# This class represents a resolved Forge module.
+#
+module Bolt
+  class ModuleInstaller
+    class Puppetfile
+      class ForgeModule
+        NAME_REGEX    = %r{\A[a-z][a-z0-9_]*[-/](?<name>[a-z][a-z0-9_]*)\z}.freeze
+        REQUIRED_KEYS = Set.new(%w[name version]).freeze
+
+        attr_reader :full_name, :name, :semantic_version, :type, :version
+
+        def initialize(name, version)
+          @full_name, @name           = parse_name(name)
+          @version, @semantic_version = parse_version(version)
+          @type                       = :forge
+        end
+
+        def self.from_hash(hash)
+          new(hash['name'], hash['version'])
+        end
+
+        def self.implements?(hash)
+          REQUIRED_KEYS == hash.keys.to_set
+        end
+
+        # Formats the full name and extracts the module name.
+        #
+        private def parse_name(name)
+          unless (match = name.match(NAME_REGEX))
+            raise Bolt::ValidationError,
+                  "Invalid name for Forge module : #{name}. Name must match 'owner/name', "\
```

Comment:
```suggestion
                  "Invalid name for Forge module: #{name}. Name must match 'owner/name', "\
```

Reply:
Wait don't we already do this in the ForgeSpecs class? How would we get here and still have this name be invalid?